pkg/kv/store_message.go

```diff
@@ -77,12 +77,15 @@ type PrimaryIterator struct {
 	err     error
 }
 
-func NewPrimaryIterator(ctx context.Context, store Store, msgType protoreflect.MessageType, partitionKey, prefix, after string) (*PrimaryIterator, error) {
+func NewPrimaryIterator(ctx context.Context, store Store, msgType protoreflect.MessageType, partitionKey, prefix, after string, skip bool) (*PrimaryIterator, error) {
```

Comment:
This signature is getting too confusing and is lacking documentation. I wouldn't know how to use it.

Reply:
skipFirst?

Reply:
I think the implementation is `skipFirstIfTheFirstValueEqualsToAfter`, which also means `after` isn't really `"After"`.

Reply:
added documentation